indexer.ts

```diff
@@ -0,0 +1,743 @@
+      ["id"],
+      []
+    );
+    if (data.admins) {
+      for (const admin in data.admins) {
+        await context.db.Account.upsert({ id: admin }, ["id"], []);
+        let factory_admin = {
+          pot_factory_id: receiverId,
+          admin_id: admin,
+        };
+        await context.db.PotFactoryAdmin.insert(factory_admin);
+      }
+    }
+
+    if (data.whitelisted_deployers) {
+      for (const deployer in data.whitelisted_deployers) {
+        await context.db.Account.upsert({ id: deployer }, ["id"], []);
+        let factory_deployer = {
+          pot_factory_id: receiverId,
+          whitelisted_deployer_id: deployer,
+        };
+        await context.db.PotFactoryWhitelistedDeployer.insert(factory_deployer);
+      }
+    }
+
+    const deploy_time = new Date(
+      Number(BigInt(block.header().timestampNanosec) / BigInt(1000000))
+    );
+    const factory = {
+      id: receiverId,
+      owner_id: data.owner,
+      deployed_at: deploy_time,
+      source_metadata: JSON.stringify(data.source_metadata),
+      protocol_fee_basis_points: data.protocol_fee_basis_points,
+      protocol_fee_recipient_account: data.protocol_fee_recipient_account,
+      require_whitelist: data.require_whitelist,
+    };
+    console.log("factory..", factory);
+    await context.db.PotFactory.insert(factory);
+  }
+
+  // function tracks registry contracts, where projects are registered
+  async function handleRegistry(args, receiverId, signerId, receiptId) {
+    let data = JSON.parse(args);
+    console.log("new Registry data::", { ...data }, receiverId);
+
+    if (data.admins) {
+      for (const admin in data.admins) {
+        await context.db.Account.upsert({ id: admin }, ["id"], []);
+        let list_admin = {
+          list_id: receiverId,
+          admin_id: admin,
+        };
+        await context.db.ListAdmin.insert(list_admin);
+      }
+    }
+
+    let regv = {
+      id: receiverId,
+      owner_id: data.owner,
+      default_registration_status: "Approved", // the first registry contract has approved as default, and the later changed through the admin set function call, which we also listen to, so it should self correct.
+      name: receiverId.split(".")[0],
+      tx_hash: receiptId,
+    };
+
+
+    await context.db.Account.upsert({ id: data.owner }, ["id"], []);
+
+    await context.db.List.insert(regv);
+  }
+
+  // function tracks register function calls on the registry conract
+  async function handleNewProject(args, receiverId, signerId, receiptId) {
+    let data = JSON.parse(args);
+    console.log("new Project data::", { ...data }, receiverId);
+    let registry = (await context.db.List.select({ id: receiverId }))[0]; // fetch the registry so as to get default
+    let reg = {
+      registrant_id: data._project_id || signerId,
+      status: registry.default_registration_status,
+      submitted_at: new Date(
+        Number(BigInt(block.header().timestampNanosec) / BigInt(1000000))
+      ),
```

Comment:
could get these from function result (ProjectExternal) if desired

Reply:
yes, but except there is a chance these data source might be incorrect, i wanted to use them since they're readily available as decoding function results take extra steps including looping through receipts.

Reply:
the thing is also that if it is registered by admin, it will be approved by default (at least that is the current functionality). i think it's probably better to take it from the decoded SuccessValue

Also as I noted elsewhere this morning, since the new Lists contract will be up very soon (ideally by EOD today) and existing Potlock registrations will be migrated to this contract via its `register` method, let's not worry about indexing the legacy Registry contract